### DISCUSSION TITLE
PIP Mode for WebRTC video calls in iOS

### DISCUSSION THREAD

**Discussion by suneelreddymunagala** [ant-media/Ant-Media-Server#5934] [Ideas]
@mekya, I started working on the PIP functionality for IOS.  I finished my rendering with RTCMTLVideoView. Then I realised what I'd done wrong: MTKView/GLKView aren't supported in pip mode. Now I intend to Create frame render to display user's video instead of RTCMTLVideoView/RTCEAGLVideoView . But I can't proceed since I can't convert RTCVideoTrack to RTCVideoFrame. Any code fragments or branches where we can discover the idea/solution would be really appreciated.

  /*
             1. get videoTrackObject
             2. convert RTCVideoTrack to RTCVideoFrame
             3. create pixelBuffer using RTCVideoFrame
             4. create CMSampleBuffer using CVPixelBuffer
             5. update in the localView/remoteview UI using cmSampleBuffer 
             6. implement PIP related methods.
             */
             
  
I referred following links to achieve the pip for video calls:
[pip for webrtc video calls in ios](https://stackoverflow.com/questions/71419635/how-to-add-picture-in-picture-pip-for-webrtc-video-calls-in-ios-swift)
[adopting_picture_in_picture_for_video_calls](https://developer.apple.com/documentation/avkit/adopting_picture_in_picture_for_video_calls)
[accessing_the_camera_while_multitasking_on_ipad](https://developer.apple.com/documentation/avkit/accessing_the_camera_while_multitasking_on_ipad)

**Comment by suneelreddymunagala** [comment]
@mekya , Any idea on this  ?

**Reply by mekya** [reply]
Hi @suneelreddymunagala ,
Thank you for the reminder. 
I don't have the opportunity to take a look at this issue. I've focused on delivering somethings to some customers. 
I try to take a look at in my convenient time. 

Alternatively, any help from community is greatly appreciated.

Thank you
Oguz
